# A: 1-4 and 7-8, D: 1-2 completed here
# A: 5-6, B: 1-2, C: 1a-1d, E: 1-5 completed in PDF

# A. Calculations

In [1]:
import numpy as np
import math
import matplotlib.pyplot as plt
from metpy.units import units

In [2]:
Ts_range_C = np.arange(-50.0, 51.0, 1.0)
Ts_range_K = Ts_range_C + 273.15
Ts_fire = 700.0

# 1)

# Wiens Law

$$
\lambda_{\text{max}} = \frac{2897}{T}
$$

The assumptions that go into Wien's Law include:
- The fire is behaving as a blackbody
- The surface is Lambertian
- This law is a result of experimental results
- The extreme value is set to 0 in the derivative used to yield this law

(Source: lecture slides)

# Calculate wavelength of maximum monocrohomatic radiance

In [3]:
lambda_max = (2897*units.K*units.micrometer)/(700*units.K)

In [4]:
print('the wavelength of maximum monochromatic radiance is ' + str(lambda_max) + 's')

the wavelength of maximum monochromatic radiance is 4.138571428571429 micrometers


# 2)

# Wiens Law

$$
\lambda_{\text{max}} = \frac{2897}{T}
$$

# Find maximum radiance in the range -50 deg C to 50 deg C using Wien's Law

In [5]:
lambda_max_range = 2897/Ts_range_K

In [6]:
print(lambda_max_range.min(), lambda_max_range.max())

8.964876992108929 12.982298902083802


In [7]:
print('the wavelength range of max radiance is ' + str(lambda_max_range.min()) + ' micrometers to ' + str(lambda_max_range.max()) + ' micrometers')

the wavelength range of max radiance is 8.964876992108929 micrometers to 12.982298902083802 micrometers


# 3)

# Stefan-Boltzmann Law

$$
F = \sigma \cdot T^4
$$

In [8]:
F_range = 5.67E-8 * (Ts_range_K**4)

In [9]:
print(F_range.min(), F_range.max())

140.5952419953945 618.3006455416394


In [10]:
print('the value range of total exitance is ' + str(F_range.min()) + ' W/m^2 to ' + str(F_range.max()) + ' W/m^2')

the value range of total exitance is 140.5952419953945 W/m^2 to 618.3006455416394 W/m^2


# 4)

# Stefan-Boltzmann Law

$$
F = \sigma \cdot T^4
$$

In [11]:
F_fire = 5.67E-8 * (Ts_fire**4)
F_fire

13613.67

In [12]:
print('the total exitance emitted from the surface in the fire zone if it were to behave as a blackbody would be ' + str(F_fire) + ' W/m^2')

the total exitance emitted from the surface in the fire zone if it were to behave as a blackbody would be 13613.67 W/m^2


# 5)
# Completed in PDF document

# 6)

# Completed in PDF document

# 7)

# Stefan-Boltzmann Law

$$
F = \sigma \cdot T^4
$$

# Multiply RHS of equation with emissivity upper and lower bounds

In [13]:
F_lower = 5.67E-8 * .7 * (Ts_fire**4)
F_upper = 5.67E-8 * .9 * (Ts_fire**4)

In [14]:
F_lower, F_upper

(9529.568999999998, 12252.303)

In [15]:
F_upper-F_lower

2722.734000000002

With a lower bound of 9529.57 W/m^2 and an upper bound of 12252.30 W/m^2, there is an uncertainty of 2722.73 W/m^2 estimated total exitance. 

# 8)

# The Planck Function


$$
B_\lambda = ({c_1}\cdot{\lambda^-5})\frac{1}{\pi(e^{\frac{c_2}{\lambda \cdot T}} - 1)}
$$

In [16]:
from climlab.utils.thermo import Planck_wavelength

/knight/mamba_aug23/envs/jan24_env/lib/python3.11/site-packages/climlab/convection/akmaev_adjustment.py:142: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  Akmaev_adjustment = jit(signature_or_function=Akmaev_adjustment)


In [17]:
B_lambda1_t1 = Planck_wavelength(3.9E-6, 600)
B_lambda1_t2 = Planck_wavelength(3.9E-6, 800)
B_lambda2_t1 = Planck_wavelength(11.0E-6, 600)
B_lambda2_t2 = Planck_wavelength(11.0E-6, 800)

In [18]:
print('at 600 K and 3.9 micrometers, the spectral emitted radiance is ' + str(B_lambda1_t1) + ' W/m^2/sr')
print('at 800 K and 3.9 micrometers, the spectral emitted radiance is ' + str(B_lambda1_t2) + ' W/m^2/sr')
print('at 600 K and 11 micrometers, the spectral emitted radiance is ' + str(B_lambda2_t1) + ' W/m^2/sr')
print('at 800 K and 11 micrometers, the spectral emitted radiance is ' + str(B_lambda2_t2) + ' W/m^2/sr')

at 600 K and 3.9 micrometers, the spectral emitted radiance is 282632223.48326325 W/m^2/sr
at 800 K and 3.9 micrometers, the spectral emitted radiance is 1324975971.471591 W/m^2/sr
at 600 K and 11 micrometers, the spectral emitted radiance is 94257475.67036898 W/m^2/sr
at 800 K and 11 micrometers, the spectral emitted radiance is 179095650.33107483 W/m^2/sr


The 3.9 μm band captures much higher monochromatic radiances for both 600 and 800 K than the 11.0 μm band. So, yes, the 3.9 μm band is much more sensitive to the hottest parts of fire pixels than the 11.0 μm band, due to its shorter wavelength.

# D. Readings and Questions

# 1.

$$
F_{TOA} = (F_{SUN} \cdot R_{SUN}^2) \cdot R_{SE}^{-2}
$$

# Convert the sun earth distance in astronomical units to meters

In [19]:
V_S = 1.0816E11  # venus
E_S = 1.496E11 # earth 
M_S = 2.27987E11 # mars

# Sun radius

In [20]:
R = 6.96E8

# Planetary albedos and constants

In [33]:
A_V = .75 # venus
A_E = .3 # earth
A_M = .16 # mars
T_sun = 5773 # sun temperature (gotten from space.com)
sigma = 5.67E-8 # stefan-boltzmann constant

# Stefan-Boltzmann Equation

$$
F_{TOA} = \sigma \cdot T^4
$$

# F_TOA

$$
F_{TOA} = \sigma \cdot T_{SUN}^4 \cdot R^2 \cdot {D}^{-2}
$$

# Calculate TOA irradiance for each planet

In [34]:
F_TOA_V = (sigma)*(T_sun**4) # venus
F_TOA_E = (sigma)*(T_sun**4) # earth
F_TOA_M = (sigma)*(T_sun**4) # mars

In [35]:
F_TOA_V_SCALED = (F_TOA_V * (R**2)) / (V_S**2) # venus
F_TOA_E_SCALED = (F_TOA_E * (R**2)) / (E_S**2) # earth
F_TOA_M_SCALED = (F_TOA_M * (R**2)) / (M_S**2) # mars

In [36]:
REF_V = A_V*F_TOA_V_SCALED # venus
REF_E = A_E*F_TOA_E_SCALED # earth
REF_M = A_M*F_TOA_M_SCALED # mars

In [37]:
ABS_V = F_TOA_V_SCALED - REF_V # venus
ABS_E = F_TOA_E_SCALED - REF_E # earth
ABS_M = F_TOA_M_SCALED - REF_M # mars

In [38]:
print('reflected TOA solar irradiance venus: ' + str(REF_V) + ' W/m^2')
print('absorbed TOA solar irradiance venus: ' + str(ABS_V) + ' W/m^2')
print('reflected TOA solar irradiance earth: ' + str(REF_E) + ' W/m^2')
print('absorbed TOA solar irradiance earth: ' + str(ABS_E) + ' W/m^2')
print('reflected TOA solar irradiance mars: ' + str(REF_M) + ' W/m^2')
print('absorbed TOA solar irradiance mars: ' + str(ABS_M) + ' W/m^2')

reflected TOA solar irradiance venus: 1955.8505851061395 W/m^2
absorbed TOA solar irradiance venus: 651.9501950353797 W/m^2
reflected TOA solar irradiance earth: 408.9458786831596 W/m^2
absorbed TOA solar irradiance earth: 954.2070502607057 W/m^2
reflected TOA solar irradiance mars: 93.90907936728667 W/m^2
absorbed TOA solar irradiance mars: 493.022666678255 W/m^2


# 2.

# Equation and rearranged equation

$$
S_0 \cdot 1/4 \cdot (1-\alpha) = \sigma \cdot T_e^4
$$

$$
T_e = \left( \frac{S_0 \cdot \frac{1}{4} \cdot (1 - \alpha)}{\sigma} \right)^{\frac{1}{4}}
$$

In [30]:
S0_V = F_TOA_V # venus
S0_E = F_TOA_E # earth
S0_M = F_TOA_M # mars

In [31]:
T_E_V = (((S0_V)*(1/4)*(1-A_V))/(sigma))**(1/4) # venus
T_E_E = (((S0_E)*(1/4)*(1-A_E))/(sigma))**(1/4) # earth
T_E_M = (((S0_M)*(1/4)*(1-A_M))/(sigma))**(1/4) # mars

In [32]:
print('without an atmosphere, venus surface temp is ' + str(T_E_V) + ' K')
print('without an atmosphere, earth surface temp is ' + str(T_E_E) + ' K')
print('without an atmosphere, mars surface temp is ' + str(T_E_M) + ' K')

without an atmosphere, venus surface temp is 2889.0 K
without an atmosphere, earth surface temp is 3737.12005408494 K
without an atmosphere, mars surface temp is 3911.401168732224 K


# From lecture:

# venus surface temp: 467 deg C = 740 K
# earth surface temp: 15 deg C = 288 K
# mars surface temp: -63 deg C = 210 K

# These results show that in general, atmospheres work to make planets warmer at the surface! The exception is Mars, which does not have a very thick atmosphere in the first place. 